megamek/src/megamek/server/GameManager.java

```diff
@@ -20268,6 +20285,54 @@ private Vector<Report> resolveControl(Entity e) {
         return vReport;
     }
 
+    /**
+     * Processes pending control roll events, merging triggers that would be multiple rolls in standard
+     * rules but are combined into a single roll under the Advanced Atmospheric Control Rolls rule.
+     *
+     * @param e         The entity whose pending rolls are being processed
+     * @param rolls     List of rolls to be added to
+     * @param reasons   Text representing the reason for the rolls
+     */
+    void addControlWithAdvAtmospheric(Entity e, Vector<PilotingRollData> rolls, StringBuilder reasons) {
+        // The August 2024 errata changed this rule to only trigger one control roll per
+        // round regardless of how many crits or thresholds occurred. As a result, the rolls
+        // need to be combined here at the end phase because not all the information is known
+        // when the individual rolls are first added throughout the round and previously added
+        // rolls can't easily be modified.
+        PilotingRollData atmosphericControlRoll = null;
+        for (Enumeration<PilotingRollData> j = game.getControlRolls(); j.hasMoreElements(); ) {
+            final PilotingRollData modifier = j.nextElement();
+            if (modifier.getEntityId() != e.getId()) {
+                continue;
+            }
+            if (Stream.of("threshold", "avionics hit", "critical hit").anyMatch(s -> modifier.getDesc().contains(s))) {
+                if (atmosphericControlRoll == null) {
+                    atmosphericControlRoll = modifier;
+                } else {
+                    // Modify the description of the pending atmospheric control roll instead of adding another one
+                    if (!reasons.isEmpty()) {
+                        reasons.append("; ");
+                    }
+                    reasons.append(modifier.getPlainDesc());
+                }
+            } else {
+                // Not an atmospheric control roll under the new rules, so treat normally
+                rolls.addElement(modifier);
+                if (!reasons.isEmpty()) {
+                    reasons.append("; ");
+                }
+                reasons.append(modifier.getPlainDesc());
+              }
```

Comment:
Will there ever be any control rolls that fall into this condition? Or at this phase of the game (end phase) will all control rolls be thresh/highest thresh/avionics/control?